src/app/content/components/ContentExcerpt.tsx

```diff
@@ -1,28 +1,52 @@
+import flow from 'lodash/fp/flow';
 import React from 'react';
+import { useSelector } from 'react-redux';
 import styled from 'styled-components/macro';
 import { bodyCopyRegularStyle } from '../../components/Typography';
-import addTargetBlankToLinks from '../utils/addTargetBlankToLinks';
+import { book } from '../selectors';
+import { findArchiveTreeNodeById } from '../utils/archiveTreeUtils';
+import { addTargetBlankToLinks, fixRelativeURLs } from '../utils/fixContentUrls';
+import { getBookPageUrlAndParams } from '../utils/urlUtils';
 
 interface Props {
   content: string;
   className: string;
+  sourcePageId: string;
 }
 
 // tslint:disable-next-line:variable-name
 const ContentExcerpt = styled((props: Props) => {
   const {
     content,
     className,
+    sourcePageId,
     ...excerptProps
   } = props;
 
-  const fixedContent = React.useMemo(
-    () => addTargetBlankToLinks(props.content),
-    [props.content]
+  const currentBook = useSelector(book);
+
+  if (!currentBook) {
+    return null;
+  }
+
+  const sourcePage = findArchiveTreeNodeById(currentBook.tree, sourcePageId);
+
+  if (!sourcePage) {
+    return null;
+  }
+
+  const excerptSource = getBookPageUrlAndParams(
+    currentBook,
+    sourcePage
   );
 
+  const fixedContent = React.useMemo(() => flow(
+    addTargetBlankToLinks,
+    (newContent) => fixRelativeURLs(newContent, excerptSource.url)),
```

Comment:
add the call for this function here instead of on line 49, otherwise you're not memoizing the logic